src/staging.c

```diff
@@ -370,6 +370,11 @@ enum swupd_code staging_install_all_files(struct list *files, struct manifest *m
 	int complete = 0;
 	unsigned int list_length = list_len(files);
 
+	if (!list_is_sorted(files, cmp_file_filename_is_deleted)) {
+		warn("List of files to install is not sorted - fixing\n");
+		files = list_sort(files, cmp_file_filename_is_deleted);
+	}
```

Comment:
This warning seems not useful to end users, more of a developer warning that somehow we got here with an unsorted list. I think getting to the point where we don't sort all over is a good goal so the logic is a step in the right direction.

Reply:
You are correct. I didn't get concerned with the warning because as this is never suppose to happen, users are not suppose to see that. But I see your point.
The reason why I didn't use debug() is because I wanted this to break our tests if it ever happen.

I was wondering other approaches we could take here. I know some projects that uses assert() for those cases. There are other areas of the code I think we could start using this, but I remember that some people in the past didn't like the idea. Do you have any opinion about that?

Reply:
assert in swupd for things that are not fatal isn't something I want to add. If it is only an assert when built for running our tests and disabled in production that could work.

Reply:
if this is a message that is really not supposed to be printed as you mentioned @otaviobp, you could just rephrase it in a way that has so meaning to the user just in the case it does get printed for some reason. Then we can use that message to catch errors in the code with tests.

An example of a message would be:
"Preparing files to be installed..."

Reply:
I am replacing for debug in this PR. Let's try to find a good solution on how to fix this kind of problems. There are multiple places in code where we do something similar, so we could have a nice approach to test that.